projects/components/src/action-menu/action-menu.component.ts

```diff
@@ -0,0 +1,194 @@
+/*!
+ * Copyright 2020 VMware, Inc.
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+import { Component, Input } from '@angular/core';
+import { ActionDisplayConfig, ActionItem, ActionStyling, ActionType, TextIcon } from '../common/interfaces';
+
+/**
+ * Renders actions in screens containing grids, cards and details container
+ */
+@Component({
+    selector: 'vcd-action-menu',
+    templateUrl: './action-menu.component.html',
+    styleUrls: ['./action-menu.component.scss'],
+})
+export class ActionMenuComponent<R, T> {
+    @Input() entityUrn: string;
+
+    /**
+     * Content of the action menu dropdown trigger button. Used when {@link #actionDisplayConfig} styling is
+     * {@link ActionStyling.DROPDOWN}
+     */
+    @Input() btnText: string = null;
+
+    /**
+     * Used for disabling the menu bar or menu dropdown
+     */
+    @Input() disabled = false;
+
+    /**
+     * A switch to hide static actions. For example, Used when we want to display only contextual actions per row or
+     * card
+     */
+    @Input() showStaticActions = true;
+
+    /**
+     * List of selected entities required for contextual actions
+     */
+    @Input() selectedEntities: R[];
+
+    /**
+     * How the static actions get displayed
+     */
+    @Input() staticActionStyling: ActionStyling = ActionStyling.INLINE;
+
+    /**
+     * List of actions that will be given by the calling component
+     */
+    private _actions: ActionItem<R, T>[] = [];
+    @Input() set actions(actions: ActionItem<R, T>[]) {
+        this._actions.length = 0;
+        actions.forEach(action => {
+            if (!action.actionType) {
+                action.actionType = ActionType.CONTEXTUAL;
+            }
+            this._actions.push(action);
+        });
+    }
+    get actions(): ActionItem<R, T>[] {
+        return this._actions;
+    }
+
+    /**
+     * Display configuration of static and contextual actions
+     */
+    private _actionDisplayConfig: ActionDisplayConfig = {
+        contextual: {
+            featuredCount: 0,
+            styling: ActionStyling.INLINE,
+            buttonContents: TextIcon.TEXT,
+        },
+        staticActionStyling: ActionStyling.INLINE,
+    };
+    @Input() set actionDisplayConfig(config: ActionDisplayConfig) {
+        if (!config) {
```

Comment:
Why is this null check required? Should it default to something when null is passed in? Please document this behavior